### ISSUE TITLE
copier do OPTIMIZE DEDUPLICATE after copying data to non-Replicated table

### ISSUE THREAD

**Issue by filimonov** [ClickHouse/ClickHouse#17966]
It can be non-desired because duplicates can exist naturally in the data, or the engine can be Summing / Aggregating.

For summing OPTIMIZE DEDUPLICATE doesn't help:

```sql
create table x (id UInt8, val UInt8) engine = SummingMergeTree ORDER BY id;
insert into x values (1,1);
insert into x values (1,1);
optimize table x deduplicate;
select * from x;
┌─id─┬─val─┐
│  1 │   2 │
└────┴─────┘
```

We should instead drop the partition and reconstruct it from chunks once again (in case of errors which are very unlikely when doing local ATTACH PARTITION FROM).

**Comment by den-crane** [comment]
And for example it could be valid duplicates in plain MT tables.

**Comment by filimonov** [comment]
Make it switchable maybe?

**Comment by nikitamikhaylov** [comment]
Will be fixed here #23518
